homeassistant/components/litterrobot/switch.py

```diff
@@ -1,76 +1,86 @@
 
 from .const import DOMAIN
-from .entity import LitterRobotConfigEntity
+from .entity import LitterRobotConfigEntity, _RobotT
 from .hub import LitterRobotHub
 
 
-class LitterRobotNightLightModeSwitch(
-    LitterRobotConfigEntity[LitterRobot], SwitchEntity
-):
-    """Litter-Robot Night Light Mode Switch."""
-
-    @property
-    def is_on(self) -> bool | None:
-        """Return true if switch is on."""
-        if self._refresh_callback is not None:
-            return self._assumed_state
-        return self.robot.night_light_mode_enabled
-
-    @property
-    def icon(self) -> str:
-        """Return the icon."""
-        return "mdi:lightbulb-on" if self.is_on else "mdi:lightbulb-off"
+@dataclass
+class RobotSwitchEntityDescription(SwitchEntityDescription, Generic[_RobotT]):
+    """A class that describes robot switch entities."""
+
+    icons: tuple[str | None, str | None] = (None, None)
+    set_fn: Callable[
+        [_RobotT], Callable[[bool], Coroutine[Any, Any, bool]]
+    ] | None = None
+
+
+ROBOT_SWITCHES = [
+    RobotSwitchEntityDescription[Union[LitterRobot, FeederRobot]](
+        key="night_light_mode_enabled",
+        name="Night Light Mode",
+        icons=("mdi:lightbulb-on", "mdi:lightbulb-off"),
+        set_fn=lambda robot: robot.set_night_light,
+    ),
+    RobotSwitchEntityDescription[Union[LitterRobot, FeederRobot]](
+        key="panel_lock_enabled",
+        name="Panel Lockout",
+        icons=("mdi:lock", "mdi:lock-open"),
+        set_fn=lambda robot: robot.set_panel_lockout,
+    ),
+]
 
-    async def async_turn_on(self, **kwargs: Any) -> None:
-        """Turn the switch on."""
-        await self.perform_action_and_assume_state(self.robot.set_night_light, True)
 
-    async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the switch off."""
-        await self.perform_action_and_assume_state(self.robot.set_night_light, False)
+class RobotSwitchEntity(LitterRobotConfigEntity[_RobotT], SwitchEntity):
+    """Litter-Robot switch entity."""
 
+    entity_description: RobotSwitchEntityDescription[_RobotT]
 
-class LitterRobotPanelLockoutSwitch(LitterRobotConfigEntity[LitterRobot], SwitchEntity):
-    """Litter-Robot Panel Lockout Switch."""
+    def __init__(
+        self,
+        robot: _RobotT,
+        hub: LitterRobotHub,
+        description: RobotSwitchEntityDescription[_RobotT],
+    ) -> None:
+        """Initialize a Litter-Robot switch entity."""
+        assert description.name
+        super().__init__(robot, description.name, hub)
+        self.entity_description = description
 
     @property
     def is_on(self) -> bool | None:
         """Return true if switch is on."""
         if self._refresh_callback is not None:
             return self._assumed_state
-        return self.robot.panel_lock_enabled
+        return bool(getattr(self.robot, self.entity_description.key))
 
     @property
-    def icon(self) -> str:
+    def icon(self) -> str | None:
```

Comment:
If it's a required key, you can keep the old return:
```suggestion
    def icon(self) -> str:
```